### PR TITLE
boards: arm: swan_r5:add 3v3 regulator

### PR DESCRIPTION
**Pull request by bsatrom**
Add configuration to turn on the 3v3 regulator on pin GPIOE4

Signed-off-by: Brandon Satrom <bsatrom+gh@gmail.com>

### CODE REVIEW

File: boards/arm/swan_r5/swan_r5.dts [hunk 1]
```diff
@@ -37,14 +37,23 @@
 	};
 
 	zephyr,user {
-		pull-up-gpios = <&gpioe 4 GPIO_PULL_UP>;
-		no-pull-gpios = <&gpioe 6 GPIO_PUSH_PULL>;
+		push-pull-gpios = <&gpioe 6 GPIO_PUSH_PULL>,
+						<&gpioc 6 GPIO_PUSH_PULL>;
```
Comment (gmarull) [inline boards/arm/swan_r5/swan_r5.dts:41]: nit: please fix alignment here

File: boards/arm/swan_r5/swan_r5.dts [hunk 2]
```diff
@@ -115,7 +124,7 @@
 	status = "okay";
 };
 
-&spi1 {
+arduino_spi: &spi1 {
```
Comment (gmarull) [inline boards/arm/swan_r5/swan_r5.dts:127]: these changes are not related: please add them to a separate commit

File: boards/arm/swan_r5/board.c [hunk 3]
```diff
@@ -9,20 +9,20 @@
...
 static int board_swan_init(const struct device *dev)
 {
-	const struct gpio_dt_spec gpio4 =
-		GPIO_DT_SPEC_GET(DT_PATH(zephyr_user), pull_up_gpios);
-	const struct gpio_dt_spec gpio6 =
-		GPIO_DT_SPEC_GET(DT_PATH(zephyr_user), no_pull_gpios);
+	const struct gpio_dt_spec gpioe6 =
+		GPIO_DT_SPEC_GET_BY_IDX(DT_PATH(zephyr_user), push_pull_gpios, 0);
+	const struct gpio_dt_spec gpioc6 =
+		GPIO_DT_SPEC_GET_BY_IDX(DT_PATH(zephyr_user), push_pull_gpios, 1);
```
Comment (gmarull) [inline boards/arm/swan_r5/board.c:15]: suggestion: rename such variables to not include gpio pin information, this code is generic in the end.

File: boards/arm/swan_r5/board.c [hunk 4]
```diff
@@ -9,20 +9,20 @@
...
-	    !device_is_ready(gpio6.port)) {
+	if (!!device_is_ready(gpioe6.port) ||
+	    !device_is_ready(gpioc6.port)) {
 		return -ENODEV;
 	}
 
-	(void)gpio_pin_configure_dt(&gpio4, (GPIO_PUSH_PULL | GPIO_SPEED_LOW));
-	(void)gpio_pin_configure_dt(&gpio6, (GPIO_NOPULL | GPIO_MODE_ANALOG));
+	(void)gpio_pin_configure_dt(&gpioe6, (GPIO_MODE_ANALOG | GPIO_NOPULL));
+	(void)gpio_pin_configure_dt(&gpioc6, (GPIO_MODE_ANALOG | GPIO_NOPULL));
```
Comment (gmarull) [inline boards/arm/swan_r5/board.c:25]: these are not standard GPIO properties... I guess they are indirectly included from a HAL... how did this code managed to work/get merged...?

**Review by gmarull — CHANGES_REQUESTED** [review]

**Comment by bsatrom** [comment]
Sorry for the delay @gmarull! I will look into addressing these items now. Appreciate the review!

File: boards/arm/swan_r5/board.c (hunk 4 above)
Comment (bsatrom) [reply boards/arm/swan_r5/board.c:25]: Yes, I am able to build with these, so I assume the STM32 HAL is being pulled in. I am using these because I couldn't find an equivalent set of flags here: https://docs.zephyrproject.org/latest/hardware/peripherals/gpio.html. Please LMK if I am missing something.

File: boards/arm/swan_r5/swan_r5.dts [hunk 5]
```diff
@@ -37,14 +37,22 @@
...
 	};
 
 	aliases {
 		led0 = &led0;
 		sw0 = &user_button;
 	};
+
+	pwr_3v3: pwr-3v3-ctrl {
+		compatible = "regulator-fixed-sync", "regulator-fixed";
+		label = "pwr-3v3-ctrl";
```
Comment (galak) [inline boards/arm/swan_r5/swan_r5.dts:50]: remove `label` property

**Review by galak — CHANGES_REQUESTED** [review]
remove label property from devicetree

File: boards/arm/swan_r5/board.c (hunk 4 above)
Comment (bsatrom) [reply boards/arm/swan_r5/board.c]: @gmarull & @galak any guidance here?

Comment (gmarull) [reply boards/arm/swan_r5/board.c]: GPIO API has documented flags, any other value will lead to UB. It's unfortunate that we indirectly pull all such crap from HALs. I hope we write some rules for HALs some day :-)

Comment (bsatrom) [reply boards/arm/swan_r5/board.c]: @gmarull understood. Beyond that, can you help me identify which of the documented and supported GPIO flags would help me here? I am using those above from the ST HAL because I didn't find an equivalent in the link above...

Comment (gmarull) [reply boards/arm/swan_r5/board.c]: What is this trying to do? maybe `GPIO_DISCONNECTED`?
